lib/instance.js

```diff
@@ -322,8 +322,11 @@ Instance.prototype.set = function(key, value, options) { // testhint options:non
           // If attribute is not in model definition, return
           if (!this._isAttribute(key)) {
             if (key.indexOf('.') > -1 && this.Model._isJsonAttribute(key.split('.')[0])) {
-              Dottie.set(this.dataValues, key, value);
-              this.changed(key, true);
+              var previousDottieValue = Dottie.get(this.dataValues, key);
+              if (!_.isEqual(previousDottieValue, value)) {
```

Comment:
Hmm, i'm not sure, the overhead is likely a function call and two type calls (without looking at the lodash code), likely negligible, thoughts @janmeier?